packages/controllers/src/snaps/SnapController.ts

```diff
@@ -424,13 +487,43 @@ export class SnapController extends BaseController<
       this._onUnresponsiveSnap,
     );
 
-    this._snapsBeingAdded = new Map();
-    this._maxIdleTime = maxIdleTime;
-    this._maxRequestTime = maxRequestTime;
-    this._idleTimeCheckInterval = idleTimeCheckInterval;
-    this._pollForLastRequestStatus();
-    this._lastRequestMap = new Map();
-    this._rpcHandlerMap = new Map();
+    this.registerMessageHandlers();
+  }
+
+  /**
+   * Constructor helper for registering the controller's messaging system
+   * actions.
+   */
+  private registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
```

Comment:
some code golf

```typescript
['add', 'get', 'getRpcMessageHandler', 'getSnapState', 'has', 'updateSnapState'].forEach((functionName) => {
 this.messagingSystem.registerActionHandler(
      `${controllerName}:${functionName}`,
      (...args) => this[functionName](...args),
    );
})

```

Reply:
This is the way. This function should go in `@metamask/controllers`.